android/src/main/java/com/pinmi/react/printer/RNNetPrinterModule.java

```diff
@@ -59,11 +59,13 @@ public void connectPrinter(String host, Integer port, Callback successCallback,
 
     @ReactMethod
     public void connectAndSend(String host, Integer port, String data, String brand,String series, Callback successCallback, Callback errorCallback) {
-        this.adapter = NetPrinterAdapter.getInstance();
-        this.adapter.init(reactContext,  null, null);
+      new Thread(() -> {
```

Comment:
How is the response handled?

Reply:
_Catching errors and forwarding to client app._

Reply:
`Callback successCallback, Callback errorCallback` arguments are there. This change won't affect them as i just wrapped the code inside a separate thread. This was tested too.